emanote/src/Emanote/Route/R.hs

```diff
@@ -88,6 +88,14 @@ encodeRoute (R slugs) =
   let parts = Slug.unSlug <$> slugs
    in withExt @a @ft $ toString $ T.intercalate "/" (toList parts)
 
+decodeXmlRoute :: FilePath -> Maybe (R 'Xml)
```

Comment:
Maybe generalize `decodeHtmlRoute` and use that?

Reply:
I'm not exactly sure how to do that.

Reply:
`coerce` should work, eg.: https://github.com/srid/emanote/blob/84aaaac7384571576506377ccdd5182fc5662ac7/emanote/src/Emanote/Model/Type.hs#L134

Reply:
Okay, let's address this at the end.